clang/include/clang/Driver/Options.td

```diff
@@ -5523,7 +5523,11 @@ def fsycl_int_header_EQ : Joined<["-"], "fsycl-int-header=">,
 def fsycl_std_layout_kernel_params: Flag<["-"], "fsycl-std-layout-kernel-params">,
   HelpText<"Enable standard layout requirement for SYCL kernel parameters.">,
   MarshallingInfoFlag<LangOpts<"SYCLStdLayoutKernelParams">>;
-defm sycl_allow_func_ptr: OptInFFlag<"sycl-allow-func-ptr", "Allow", "Disallow", " function pointers in SYCL device.", [CC1Option,CoreOption], LangOpts<"SYCLAllowFuncPtr">>;
+defm sycl_allow_func_ptr: BoolFOption<"sycl-allow-func-ptr",
+  LangOpts<"SYCLAllowFuncPtr">, DefaultFalse,
+  PosFlag<SetTrue, [], "Allow">,
+  NegFlag<SetFalse, [], "Disallow">,
+  BothFlags<[CC1Option, CoreOption], " function pointers in SYCL device.">>;
```

Comment:
Not sure about this but should only PosFlag be -cc1 option? Saw a bunch of options in Options.td follow that format. Does using BothFlags here mean both positive and negative version can be passed to FE?

Reply:
We had this codes before https://github.com/intel/llvm/commit/a52d3d3aa6c9ceb3340f1692deec9c0244f73902

def fsycl_allow_func_ptr : Flag<["-"], "fsycl-allow-func-ptr">,
  HelpText<"Allow function pointers in SYCL device.">;
def fno_sycl_allow_func_ptr : Flag<["-"], "fno-sycl-allow-func-ptr">;

Opts.SYCLAllowFuncPtr =
        Args.hasFlag(options::OPT_fsycl_allow_func_ptr,
                     options::OPT_fno_sycl_allow_func_ptr, false);

Here Both PosFlag and NegFlag should be --cc1 option.



>>Does using BothFlags here mean both positive and negative version can be passed to FE?
Yes, correct.

Reply:
@elizabethandrews, does @smann12's explanation suffice for you?  I am not too familiar with this part of the code and I had consulted offline with @smanna12 and @mdtoguchi on this.